doc/reference/trd-userspace-readable-allow-syscalls.md

```diff
@@ -0,0 +1,183 @@
+Userspace Readable Allow System Call
+========================================
+
+**TRD:** XXX <br/>
+**Working Group:** Kernel<br/>
+**Type:** Documentary<br/>
+**Status:** Draft <br/>
+**Author:** Alistair Francis <br/>
+**Draft-Created:** June 17, 2021<br/>
+**Draft-Modified:** June 17, 2021<br/>
+**Draft-Version:** 2<br/>
+**Draft-Discuss:** tock-dev@googlegroups.com</br>
+
+Abstract
+-------------------------------
+
+This document describes the userspace readable allow system call application binary
+interface (ABI) between user space processes and the Tock kernel for 32-bit
+ARM Cortex-M and RISC-V RV32I platforms.
+
+This is an extension on the allow calls specified in
+[TRD 104](trd104-syscalls.md).
+
+1 Introduction
+==============
+
+In normal use, userspace does not access a buffer that
+has been userspace readable with the kernel with a Read-Write Allow call. This
+reading restriction is because the contents of the buffer may be in an
+intermediate state and so not consistent with expected data
+models. Ensuring every system call driver maintains consistency in the
+presence of arbitrary userspace reads is too great a programming
+burden for an unintended use case.
+
+However, there can be cases when it is necessary for userspace to be
+able to read a buffer without first revoking it from the kernel with a
+Read-Write Allow. These cases are situations when the cost of a
+Read-Write Allow system call is an unacceptable overhead for
+accessing the data.
+
+Instead, capsules that support the userspace readable allow call can communicate with
+applications without buffers needing to be re-allowed. For example a capsule
```

Comment:
The userspace readable allow system call allows userspace to have read-only access a buffer that is writeable by the kernel.